python/test/unit/runtime/test_compilation_listener.py

```diff
@@ -0,0 +1,84 @@
+import triton
+import triton.language as tl
+
+from triton.backends.compiler import GPUTarget
+from triton.compiler.compiler import ASTSource, IRSource
+
+from triton.runtime.config import TritonConfig, CompileTimes
+from typing import Any, Union
+
+import torch
+
+
+def get_add_kernel() -> triton.JITFunction:
+
+    @triton.jit
+    def add_kernel(
```

Comment:
You can just define this in the file scope and use it directly.

Reply:
That won't trigger compilation again, so we won't get a subsequent callback invocation in the test method. I'll add a comment stating this

Reply:
You can clear the in-memory cache with `add_kernel.device_caches.clear()` to trigger a cache hit.